lib/simplifyExpression/substituteScopeSearch/index.js

```diff
@@ -0,0 +1,60 @@
+const ChangeTypes = require('../../ChangeTypes');
+const Node = require('../../node');
+const TreeSearch = require('../../TreeSearch');
+
+// Searches through the tree, prioritizing deeper nodes, and substitutes
+// in-scope values for their respective expressions on a symbol node if possible.
+// Returns a Node.Status object.
+const search = TreeSearch.postOrder(scopeSubstitution);
+
+function scopeSubstitution(node, options={}) {
+  if (Node.Type.isSymbol(node)) {
+    return substituteAndSimplifySymbols(node, options);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// SUBSTITUTES
+// Returns a Node.Status object with substeps
+function substituteAndSimplifySymbols(node, options={}) {
+  if (!options.hasOwnProperty('scope')) {
+    return Node.Status.noChange(node);
+  }
+
+  if (!Node.Type.isSymbol(node)) {
+    return Node.Status.noChange(node);
+  }
+
+  let symbolName;
+  if (node.type === 'SymbolNode') {
+    symbolName = node.name;
+  }
+  else if (Node.Type.isUnaryMinus(node)) {
+    symbolName = node.args[0].name;
+  }
+
+  const scope = options.scope;
+  if (scope.hasOwnProperty(symbolName)) {
+    // when declared at top, kept getting
+    // TypeError: <nameIGaveToFunction> is not a function
```

Comment:
what is `nameIGaveToFunction` - which function?

Reply:
...this was my way of trying to communicate that, no matter what name I gave to the function that made this call, it would fail with that error message. `nameIGaveToFunction` would be substituted with whatever I tried. IIRC, I pointed this out because I thought the issue might have been with a name conflict of some kind. Doesn't appear to be the case, though.

Reply:
which is the function making the call? is it in the PR here? what's it called right now?

There's so many functions in the code haha so confused which one you're referring to!